roles/nv_gpu_install_from_commit/tasks/build.yml

```diff
@@ -67,33 +69,42 @@
   shell: oc delete -f "{{ gpu_operator_image_builder_pod }}" || true
 
-- name: delete operator image, if any
-  shell: oc delete imagestreamtag/gpu-operator-ci:gpu-operator -n gpu-operator-ci || true
+- name: prepare the operator image
+  block:
+  - name: check if the operator image exists
+    shell: "oc get imagestreamtag/gpu-operator-ci:{{ gpu_operator_image_tag }} -n gpu-operator-ci"
 
-- name: apply operator image builder pod manifest
-  shell: sed 's|{{ '{{' }} builder_secret {{ '}}' }}|{{ builder_secret.stdout }}|' "{{ gpu_operator_image_builder_pod }}" \
-       | sed 's|{{ '{{' }} gpu_operator_git_repo {{ '}}' }}|{{ gpu_operator_git_repo }}|' \
-       | sed 's|{{ '{{' }} gpu_operator_git_ref {{ '}}' }}|{{ gpu_operator_git_ref }}|' \
-       | sed 's|{{ '{{' }} gpu_operator_image_tag {{ '}}' }}|{{ gpu_operator_image_tag }}|' \
-       | oc apply -f-
-  args:
-    warn: false # don't warn about using sed here
+  rescue:
+  - debug:
+      msg: the operator image does not exist, build it
+  - name: apply operator image builder pod manifest
+    shell: sed 's|{{ '{{' }} builder_secret {{ '}}' }}|{{ builder_secret.stdout }}|' "{{ gpu_operator_image_builder_pod }}" \
+         | sed 's|{{ '{{' }} gpu_operator_git_repo {{ '}}' }}|{{ gpu_operator_git_repo }}|' \
+         | sed 's|{{ '{{' }} gpu_operator_git_ref {{ '}}' }}|{{ gpu_operator_git_ref }}|' \
+         | sed 's|{{ '{{' }} gpu_operator_image_tag {{ '}}' }}|{{ gpu_operator_image_tag }}|' \
+         | oc apply -f-
+    args:
+      warn: false # don't warn about using sed here
 
-- name: wait for the operator image to be built
-  block:
   - name: wait for the operator image to be built
-    shell: oc get pod --field-selector=status.phase=Succeeded,metadata.name=operator-image-builder-pod --no-headers -oname -n gpu-operator-ci | grep operator-image-builder-pod
-    register: wait_image_builder_pod
-    until: wait_image_builder_pod.rc != 1
-    retries: 40
-    delay: 30
-  rescue:
-  - name: get info about operator image build failure
-    command: oc get pods -n gpu-operator-ci
-    ignore_errors: true
-  - name: get logs of image build failure
-    command: oc get logs pod/operator-image-builder-pod
-    ignore_errors: true
-  - name: operator image failed to build
-    fail:
-      msg: operator image failed to build
+    block:
+    - name: wait for the operator image to be built
+      shell: oc get pod --field-selector=status.phase=Succeeded,metadata.name=operator-image-builder-pod \
+                        --no-headers \
+                        -oname \
+                        -n gpu-operator-ci \
+          | grep operator-image-builder-pod
+      register: wait_image_builder_pod
+      until: wait_image_builder_pod.rc != 1
+      retries: 40
+      delay: 30
+    rescue:
+    - name: get info about operator image build failure
+      command: oc get pods -n gpu-operator-ci
+      ignore_errors: true
+    - name: get logs of image build failure
```

Comment:
I don't see the value of this command? are we storing them or something? we should pipe the logs into a file and store them on an artifact at least. but like these I don't see the value added from this task

Reply:
it's for debugging the CI failures! as I mentioned in a slack discussion, the way I would like the tests to be (eventually) written is:
```
- block:
  - command: do a test
- rescue: # do a test failed
  - command: log whatever can help understanding why `do a test` failed
  - fail:
    - msg: failing because of previous error
```
this way, when a test fail, you have (hopefully) all the information to tell what what wrong.

some commands like this one, a few days ago, explains its failure:
```
    "stderr_lines": [
        "+ '[' 4 -gt 4 ']'",
        "+ action=deploy",
        "+ OPERATOR_GIT_REPO=https://gitlab.com/kpouget_psap/gpu-operator.git",
        "+ OPERATOR_GIT_REF=ci-testing",
        "+ OPERATOR_IMAGE_TAG=gpu-operator-cc13ea12",
        "+ OPERATOR_NAME=gpu-operator-from-source",
        "+ OPERATOR_NAMESPACE=gpu-operator-ci",
        "+ HELM_SOURCE=./deployments/gpu-operator",
        "+ '[' deploy == deploy ']'",
        "+ deploy",
        "+ cd /tmp",
        "+ rm -rf gpu-operator",
        "+ git clone https://gitlab.com/kpouget_psap/gpu-operator.git -b ci-testing --depth 1 gpu-operator",
        "Cloning into 'gpu-operator'...",
        "fatal: unable to access 'https://gitlab.com/kpouget_psap/gpu-operator.git/': The requested URL returned error: 500"
    ],
```

but `wait for the operator image to be built` will not give you any information about what went wrong. Taking the logs of the build failure will give you the explanation.

Reply:
(notice that it's a the `rescue` block, so it won't be executed as part of the normal workflow)